Updates wallet and address information. (#522)
filecoin-project/filecoin-docs: docs/mine/lotus/miner-wallets.md

---

> title: 'Lotus Miner: Wallets'

Reviewer: ```suggestion
title: 'Lotus Miner: Addresses'
```

We need to rename the whole file and create a redirect @johnnymatthews

> description: 'A miner can be configured with an owner address, a worker address, and additional control addresses. This separation of addresses allows for more secure transfers and more granual management of each address.'

Author: ```suggestion
description: 'A miner can be configured with an owner address, a worker address, and additional control addresses. These allow granularity in how the funds sent and received from the miner are managed and provide additional security to the mining operation.'
```

> breadcrumb: 'Miner Wallets'

Reviewer: ```suggestion
breadcrumb: 'Miner addresses'
```

The Lotus Miner daemon performs the operations required by the network and can use different Lotus node wallets to pay the fees or interact with the _miner actor_. Check out the [Lotus Getting Started guide](../../get-started/lotus/send-and-receive-fil.md) for more information on how to manage Lotus wallets.

::: tip
Each node can only have one wallet, but each wallet can have multiple addresses.
:::

The currently configured addresses used by a miner can be listed with:

```sh
lotus-miner actor control list
```

The different types of addresses associated with a miner are described below:

## The owner address

The owner address corresponds to a Lotus node address provided during the [miner initialization](miner-setup.md). The _owner address_ is only needed when:

- Changing the owner or the worker address in the _miner actor_.
- Withdrawing balance from the _miner actor_.
- Submit _WindowPoSts_, **unless _control addresses_ are defined and have enough balance** (continued below).

> The wallet associated with the _owner address_ is designed to be kept offline as a cold wallet since it should not be used often. It is critical to safeguard the miner's funds. In production environments, we strongly recommend using separate _owner_ and _worker_ addresses.

Author: @hsanjuan, I'm confused about this section.

> The wallet associated with the _owner address_ is designed to be kept offline as a cold wallet.

This makes sense. But if we replace _The wallet associated_ with _The address associated_ then it doesn't, since you can't have a _cold-address_. Can you?

Reviewer: I think we want to say something like `The address chosen to be the miner's _owner address_ is designed to be mostly kept offline or otherwise backed by [hardware wallet](link to ledger guide)`.

The address will need to be available to lotus in some moments when performing some of the actions but otherwise it can be exported and removed.

Author: ```suggestion
The address chosen to be the miner's _owner address_ is designed to be kept offline in _cold storage_, or backed up by a [hardware wallet](../../get-started/lotus/ledger). In production environments, we strongly recommend using separate _owner_ and _worker_ addresses.
```

The owner address can be updated with the following command:

```sh
lotus-miner actor set-owner --really-do-it <address>
```

The old and the new address must be available in the Lotus node. You can [create a new address or import an existing one](../../get-started/lotus/send-and-receive-fil.md).

## The worker address

The _worker address_ is used to send and pay for day-to-day operations performed by the miner:

- Initializing the miner on the chain.
- Changing the miner peer id or the multiaddresses.
- Interacting with market and payment channel actors.
- Signing new blocks.
- Submitting proofs, declaring faults. _WindowPoSts_ are submitted using the _worker address_ if:
  - Control addresses are not defined or do not have enough balance.
  - The _owner address_ does not have enough balance.

> Unlike the _owner wallet_, the _worker wallet_ must be a _hot_ wallet. The Lotus Miner will trigger all the necessary transactions using the Lotus node to which it is connected. The _worker address_ must have enough funds to pay for the day-to-day operations of the miner, including initialization.

Author: @hsanjuan, again, I'm confused by this terminology. Hot-wallets exist, but if there's only 1 wallet per node, then how can we have several wallets?

Reviewer: How about something like: `Unlike the _owner  address_, the address set as the miner's _worker address_ should be part of the Lotus wallet and accessible by the miner.`

Author: ```suggestion
Unlike the _owner address_, the address set as the miner's _worker address_ should be part of the Lotus local wallet and accessible by the miner. The Lotus Miner will trigger all the necessary transactions using the Lotus node to which it is connected. The _worker address_ must have enough funds to pay for the day-to-day operations of the miner, including initialization.
```

## Control addresses

_Control addresses_ can be used to submit _WindowPoSts_ proofs to the chain. _WindowPoSt_ is the mechanism through which storage is verified in Filecoin and which requires miners to submit proofs for all sectors every 24 hours. Those proofs are submitted as messages to the blockchain and therefore need to pay the respective fees.

Many mining-related actions require sending messages to the chain, but not all of those are as high-value as _WindowPoSts_. We recommended using _control addresses_ to avoid head-of-line blocking problems in congested chain conditions. [Head-of-line blocking](https://en.wikipedia.org/wiki/Head-of-line_blocking) is a performance-limiting phenomenon that occurs when the first transaction holds up a line of transactions.

Multiple _control addresses_ can be created and configured in a Lotus miner. The first _control address_ found to have enough funds to submit a _WindowPoSt_ transaction will be used. Otherwise, Lotus fails over to the _owner_ and ultimately to the _worker_ address.

To set up a _control address_:

> 1. Create a new wallet and send it some funds for gas fees:

Reviewer: Create a new address

Author: ```suggestion
1. Create a new address and send it some funds for gas fees:
```

## Managing balances

Get the balances associated with a miner wallet by calling `info`:

```bash
lotus-miner info

> Miner: t01000
> Sector Size: 2 KiB
> Byte Power:   100 KiB / 100 KiB (100.0000%)
> Actual Power: 1e+03 Ki / 1e+03 Ki (100.0000%)
>   Committed: 100 KiB
>   Proving: 100 KiB
> Below minimum power threshold, no blocks will be won
> Deals: 0, 0 B
>   Active: 0, 0 B (Verified: 0, 0 B)
>
> Miner Balance: 10582.321501530685596531 FIL
>   PreCommit:   0.000000286878768791 FIL
>   Pledge:      0.00002980232192 FIL
>   Locked:      10582.321420164834231291 FIL
>   Available:   0.000051276650676449 FIL
> Worker Balance: 49999999.999834359275302423 FIL
> Market (Escrow):  0 FIL
> Market (Locked):  0 FIL
```

In this example, the miner ID is `t01000`, it has a total balance of `10582.321501530685596531 FIL`, and an available balance of `0.000051276650676449 FIL` that can be used as collateral or to pay for the pledge. The worker balance is `49999999.999834359275302423 FIL`.

## Withdrawing funds from the Miner actor

Transfer funds from the Miner actor address to the owner address by calling `actor withdraw`:

```bash
lotus-miner actor withdraw <amount>
```

::: tip

> The owner's address will need to be available in the Lotus node and have enough funds to pay for the gas for this transaction. Cold wallets will need to be temporally imported for the operation to succeed.

Reviewer: ```suggestion
The owner's address will need to be available in the Lotus node and have enough funds to pay for the gas for this transaction. Cold addresses will need to be temporally imported for the operation to succeed.
```

(or addresses in cold wallets. But what is imported is the address, not the wallet).
